Auto create pivot permissions cleanup

mourya-33 opened the pull request:
### Feature or Bugfix
- Bugfix

### Detail
This PR is to restrict few overly permissive policy permissions on the auto created Pivot Role

### Relates
[- <URL or Ticket>](https://github.com/data-dot-all/dataall/issues/875)

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)? No
  - Is the input sanitized? N/A
  - What precautions are you taking before deserializing the data you consume? N/A
  - Is injection prevented by parametrizing queries?N/A
  - Have you ensured no `eval` or similar functions are used?N/A
- Does this PR introduce any functionality or component that requires authorization?N/A
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?N/A
  - Are you logging failed auth attempts?N/A
- Are you using or adding any cryptographic features?N/A
  - Do you use a standard proven implementations?Yes
  - Are the used keys controlled by the customer? Where are they stored?N/A
- Are you introducing any new policies/roles/users?N/A
  - Have you used the least-privilege principle? How?N/A


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

mourya-33 commented:
Tested the following

1. Linking environments
2. Unlinking environments
3. Add datasets, delete datasets
4. ML Studio and Notebooks

noah-paige commented on backend/dataall/modules/datapipelines/cdk/pivot_role_datapipelines_policy.py:
```diff
@@ -50,7 +50,7 @@ def get_statements(self):
                 ],
                 resources=[
-                    f'arn:aws:cloudformation:*:{self.account}:stack/*/*',
+                    f'arn:aws:cloudformation:*:{self.account}:stack/{self.env_resource_prefix}*/*',
```
Isn't this permission already granted in `/dataall/core/environment/cdk/pivot_role_core_policies/cloudformation.py`?

Do we need this permission then for data.all pipelines?

mourya-33 replied:
@noah-paige i removed this permission in the data pipelines as this is already covered under core policy

noah-paige commented:
Request - add changes to `pivotRole.yaml` as well

mourya-33 commented:
> Request - add changes to `pivotRole.yaml` as well

Done @noah-paige 

Changes:
1. removed DeleteApp permission for sagemaker as the Apps are not created by pivotRole
2. removed cloudformation stack permissions for pipelines, as this is already covered under the core cloudformation policy for pivot role

noah-paige commented on deploy/pivot_role/pivotRole.yaml:
```diff
@@ -213,7 +213,8 @@ Resources:
             Effect: Allow
-            Resource: '*'
+            Resource: 
+              - !Sub 'arn:aws:logs:*:${AWS::AccountId}:log-group:/${EnvironmentResourcePrefix}*'
```
In CDK Pivot Role we have the resource as `arn:aws:logs:*:{self.account}:log-group:*` but here it is different

mourya-33 replied:
I merged PutLogEvents with the "Logs" SID statement set and added a prefix for lambda function log groups under resources

noah-paige commented:
Tested CDK Created Pivot Role, AWS Deployment

- [ ] CICD Pipeline Succeeds
- [ ] Update Env Stack --> Updates Pivot Role Permissions
- [ ] Create Data.all Pipeline (CDK and CodePipeline)
- [ ] Create and Delete ML Studio User Profile

anmolsgandhi reviewed (approved):
Reviewed with @noah-paige , LGTM!
